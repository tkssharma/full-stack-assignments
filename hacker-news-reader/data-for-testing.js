export const stories = [
  {
    by: 'magoghm',
    descendants: 41,
    id: 18277531,
    kids: [18277926, 18277900, 18278074, 18277699, 18277859, 18277835, 18277864],
    score: 79,
    time: 1540236259,
    title: 'AWS CEO Jassy follows Apple in calling for retraction of Chinese spy chip story',
    type: 'story',
    url:
      'https://www.cnbc.com/2018/10/22/aws-ceo-jassy-follows-apple-calls-for-spy-chip-story-retraction.html',
  },
  {
    by: 'NN88',
    descendants: 72,
    id: 18276862,
    kids: [
      18277851,
      18277773,
      18277336,
      18277638,
      18277523,
      18277196,
      18277173,
      18277302,
      18277409,
      18277700,
      18277870,
      18277390,
      18277646,
      18277581,
      18277721,
      18277324,
      18277689,
      18277514,
      18277389,
    ],
    score: 105,
    time: 1540231727,
    title: 'The Secretive Business of Facial-Recognition Software in Retail Stores',
    type: 'story',
    url:
      'http://nymag.com/intelligencer/2018/10/retailers-are-using-facial-recognition-technology-too.html',
  },
];

// we need time to be fixed relative to now (like 2 hours ago),
// we need the number of seconds since 1970.
// Note, because the props or the way the time is displayed always changes, we can't do snapshot on this component
var time = Date.now() * 1 - 1000 * 3600 * 3;
time = Math.round(time / 1000);

export const listItemData = {
  by: 'NN88',
  kids: [
    18277581,
    18277336,
    18277409,
    18277173,
    18277196,
    18277523,
    18277390,
    18277302,
    18277324,
    18277514,
    18277389,
  ],
  score: 58,
  url:
    'http://nymag.com/intelligencer/2018/10/retailers-are-using-facial-recognition-technology-too.html',
  title: 'The Secretive Business of Facial-Recognition Software in Retail Stores',
  id: 18276862,
  type: 'story',
  time: time,
};

export const gridItemData = {
  url:
    'http://nymag.com/intelligencer/2018/10/retailers-are-using-facial-recognition-technology-too.html',
  title: 'The Secretive Business of Facial-Recognition Software in Retail Stores',
  id: 18276862,
};

export const AppData = {
  hasMoreStores: false,
  isFetching: false,
  layout: 'list',
  page: 1,
  stories: stories,
  storyIds: [18277531, 18276862],
  theme: 'dark',
};

export const NavData = {
  layout: 'list',
  theme: 'dark',
};
