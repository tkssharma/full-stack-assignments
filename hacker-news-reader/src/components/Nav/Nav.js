import React from 'react';
import PropTypes from 'prop-types';
import { layouts, themes } from 'store/app/utils';

import { Header, Spacer, NavSection, Content, Icon, Logo, ExternalLink } from './styles';

const Nav = ({ layout, theme, setLayout, setTheme }) => (
  <div>
    <Header>
      <Content>
        <NavSection>
          <ExternalLink href="https://gitconnected.com" target="_blank">
            <Logo src="https://gitconnected.com/public/meta/favicon/gc-fav.png" /> gitconnected
          </ExternalLink>
        </NavSection>
        <NavSection>
          {layout === layouts.list ? (
            <Icon onClick={() => setLayout(layouts.grid)}>
              <i className="fas fa-th-large" />
            </Icon>
          ) : (
            <Icon onClick={() => setLayout(layouts.list)}>
              <i className="fas fa-th-list" />
            </Icon>
          )}
          {theme === themes.light ? (
            <Icon onClick={() => setTheme(themes.dark)}>
              <i className="fas fa-moon" />
            </Icon>
          ) : (
            <Icon onClick={() => setTheme(themes.light)}>
              <i className="fas fa-sun" />
            </Icon>
          )}
        </NavSection>
      </Content>
    </Header>
    <Spacer />
  </div>
);

Nav.propTypes = {
  layout: PropTypes.string.isRequired,
  theme: PropTypes.string.isRequired,
  setLayout: PropTypes.func.isRequired,
  setTheme: PropTypes.func.isRequired,
};

export default Nav;
