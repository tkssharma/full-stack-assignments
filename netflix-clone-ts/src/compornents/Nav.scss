.Nav {
  position: fixed;
  top: 0;
  width: 100%;
  height: 30px;
  padding: 20px;
  z-index: 1;
  display: flex;
  justify-content: space-between;

  /* Animations */
  transition-timing-function: ease-in;
  transition: all 0.5s;

  &-black {
    background-color: #111;
  }

  &-logo {
    position: fixed;
    left: 20px;
    width: 80px;
    object-fit: contain;
  }

  &-avater {
    position: fixed;
    right: 20px;
    width: 30px;
    object-fit: contain;
  }
}
