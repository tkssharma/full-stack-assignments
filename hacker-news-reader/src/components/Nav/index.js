import { connect } from 'react-redux';
import actions from 'store/app/actions';
import Nav from './Nav';

const mapStateToProps = state => ({
  layout: state.app.layout,
  theme: state.app.theme,
});

const mapDispatchToProps = dispatch => ({
  setTheme: theme => dispatch(actions.setTheme({ theme })),
  setLayout: layout => dispatch(actions.setLayout({ layout })),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Nav);
