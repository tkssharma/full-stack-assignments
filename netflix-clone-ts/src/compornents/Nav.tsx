import React, { useState, useEffect, useCallback } from "react";
import "./Nav.scss";

type Props = {
  className?: string;
};

export const Nav = (props: Props) => {
  const [show, setShow] = useState(false);
  useEffect(() => {
    const handleShow = () => {
      if (window.scrollY > 100) {
        setShow(true);
      } else {
        setShow(false);
      }
    };

    window.addEventListener("scroll", handleShow);
    return () => {
      window.removeEventListener("scroll", handleShow);
    };
  }, []);
  return (
    <div className={`Nav ${show && "Nav-black"}`}>
      <img
        className="Nav-logo"
        src="https://upload.wikimedia.org/wikipedia/commons/thumb/0/08/Netflix_2015_logo.svg/1920px-Netflix_2015_logo.svg.png"
        alt="Netflix Logo"
      />
      <img
        className="Nav-avater"
        src="https://i.pinimg.com/originals/0d/dc/ca/0ddccae723d85a703b798a5e682c23c1.png"
        alt="Avatar"
      />
    </div>
  );
};
