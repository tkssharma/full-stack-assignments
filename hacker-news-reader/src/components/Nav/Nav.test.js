import React from 'react';
import { NavData } from '../../../data-for-testing';
import { shallow } from 'enzyme';
import Nav from './Nav';
const { layout, theme } = NavData;

describe('App Component', () => {
  let wrapper;
  const mocksetLayout = jest.fn();
  const mocksetTheme = jest.fn();
  beforeEach(() => {
    // pass the mock function as the login prop
    wrapper = shallow(
      <Nav setLayout={mocksetLayout} setTheme={mocksetTheme} layout={layout} theme={theme} />,
    );
  });

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // Note: enzyme click events do not propagate...
  it('should call the mocksetLayout function on click', () => {
    wrapper
      .find('.fa-th-large')
      .parent()
      .simulate('click');
    expect(mocksetLayout.mock.calls.length).toBe(1);
  });

  it('should call the mocksetTheme function on click', () => {
    wrapper
      .find('.fa-sun')
      .parent()
      .simulate('click');
    expect(mocksetTheme.mock.calls.length).toBe(1);
  });
});
