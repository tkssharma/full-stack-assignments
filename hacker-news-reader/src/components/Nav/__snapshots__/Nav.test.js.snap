// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    layout="list"
    setLayout={[Function]}
    setTheme={[Function]}
    theme="dark"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.header>
          <styled.div>
                    <styled.div>
                              <styled.a
                                        href="https://gitconnected.com"
                                        target="_blank"
                              >
                                        <styled.img
                                                  src="https://gitconnected.com/public/meta/favicon/gc-fav.png"
                                        />
                                         gitconnected
                              </styled.a>
                    </styled.div>
                    <styled.div>
                              <styled.div
                                        onClick={[Function]}
                              >
                                        <i
                                                  className="fas fa-th-large"
                                        />
                              </styled.div>
                              <styled.div
                                        onClick={[Function]}
                              >
                                        <i
                                                  className="fas fa-sun"
                                        />
                              </styled.div>
                    </styled.div>
          </styled.div>
</styled.header>,
        <styled.div />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.div>
            <styled.div>
                        <styled.a
                                    href="https://gitconnected.com"
                                    target="_blank"
                        >
                                    <styled.img
                                                src="https://gitconnected.com/public/meta/favicon/gc-fav.png"
                                    />
                                     gitconnected
                        </styled.a>
            </styled.div>
            <styled.div>
                        <styled.div
                                    onClick={[Function]}
                        >
                                    <i
                                                className="fas fa-th-large"
                                    />
                        </styled.div>
                        <styled.div
                                    onClick={[Function]}
                        >
                                    <i
                                                className="fas fa-sun"
                                    />
                        </styled.div>
            </styled.div>
</styled.div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <styled.a
                                href="https://gitconnected.com"
                                target="_blank"
                >
                                <styled.img
                                                src="https://gitconnected.com/public/meta/favicon/gc-fav.png"
                                />
                                 gitconnected
                </styled.a>
</styled.div>,
              <styled.div>
                <styled.div
                                onClick={[Function]}
                >
                                <i
                                                className="fas fa-th-large"
                                />
                </styled.div>
                <styled.div
                                onClick={[Function]}
                >
                                <i
                                                className="fas fa-sun"
                                />
                </styled.div>
</styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <styled.a
                  href="https://gitconnected.com"
                  target="_blank"
>
                  <styled.img
                                    src="https://gitconnected.com/public/meta/favicon/gc-fav.png"
                  />
                   gitconnected
</styled.a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.img
                      src="https://gitconnected.com/public/meta/favicon/gc-fav.png"
/>,
                    " gitconnected",
                  ],
                  "href": "https://gitconnected.com",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "src": "https://gitconnected.com/public/meta/favicon/gc-fav.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  " gitconnected",
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.div
                    onClick={[Function]}
>
                    <i
                                        className="fas fa-th-large"
                    />
</styled.div>,
                  <styled.div
                    onClick={[Function]}
>
                    <i
                                        className="fas fa-sun"
                    />
</styled.div>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <i
                      className="fas fa-th-large"
/>,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-th-large",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <i
                      className="fas fa-sun"
/>,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-sun",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.header>
            <styled.div>
                        <styled.div>
                                    <styled.a
                                                href="https://gitconnected.com"
                                                target="_blank"
                                    >
                                                <styled.img
                                                            src="https://gitconnected.com/public/meta/favicon/gc-fav.png"
                                                />
                                                 gitconnected
                                    </styled.a>
                        </styled.div>
                        <styled.div>
                                    <styled.div
                                                onClick={[Function]}
                                    >
                                                <i
                                                            className="fas fa-th-large"
                                                />
                                    </styled.div>
                                    <styled.div
                                                onClick={[Function]}
                                    >
                                                <i
                                                            className="fas fa-sun"
                                                />
                                    </styled.div>
                        </styled.div>
            </styled.div>
</styled.header>,
          <styled.div />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.div>
              <styled.div>
                            <styled.a
                                          href="https://gitconnected.com"
                                          target="_blank"
                            >
                                          <styled.img
                                                        src="https://gitconnected.com/public/meta/favicon/gc-fav.png"
                                          />
                                           gitconnected
                            </styled.a>
              </styled.div>
              <styled.div>
                            <styled.div
                                          onClick={[Function]}
                            >
                                          <i
                                                        className="fas fa-th-large"
                                          />
                            </styled.div>
                            <styled.div
                                          onClick={[Function]}
                            >
                                          <i
                                                        className="fas fa-sun"
                                          />
                            </styled.div>
              </styled.div>
</styled.div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.div>
                  <styled.a
                                    href="https://gitconnected.com"
                                    target="_blank"
                  >
                                    <styled.img
                                                      src="https://gitconnected.com/public/meta/favicon/gc-fav.png"
                                    />
                                     gitconnected
                  </styled.a>
</styled.div>,
                <styled.div>
                  <styled.div
                                    onClick={[Function]}
                  >
                                    <i
                                                      className="fas fa-th-large"
                                    />
                  </styled.div>
                  <styled.div
                                    onClick={[Function]}
                  >
                                    <i
                                                      className="fas fa-sun"
                                    />
                  </styled.div>
</styled.div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <styled.a
                    href="https://gitconnected.com"
                    target="_blank"
>
                    <styled.img
                                        src="https://gitconnected.com/public/meta/favicon/gc-fav.png"
                    />
                     gitconnected
</styled.a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <styled.img
                        src="https://gitconnected.com/public/meta/favicon/gc-fav.png"
/>,
                      " gitconnected",
                    ],
                    "href": "https://gitconnected.com",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "src": "https://gitconnected.com/public/meta/favicon/gc-fav.png",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    " gitconnected",
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.div
                      onClick={[Function]}
>
                      <i
                                            className="fas fa-th-large"
                      />
</styled.div>,
                    <styled.div
                      onClick={[Function]}
>
                      <i
                                            className="fas fa-sun"
                      />
</styled.div>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <i
                        className="fas fa-th-large"
/>,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fas fa-th-large",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <i
                        className="fas fa-sun"
/>,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fas fa-sun",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
